bundles/org.openhab.binding.goecharger/src/main/java/org/openhab/binding/goecharger/internal/handler/GoEChargerV2Handler.java

```diff
@@ -0,0 +1,349 @@
+            case POWER_L2:
+                if (goeResponse.energy == null) {
+                    return UnDefType.UNDEF;
+                }
+                return new QuantityType<>(goeResponse.energy[8] * 1000, Units.WATT);
+            case POWER_L3:
+                if (goeResponse.energy == null) {
+                    return UnDefType.UNDEF;
+                }
+                return new QuantityType<>(goeResponse.energy[9] * 1000, Units.WATT);
+            case FORCE_STATE:
+                if (goeResponse.forceState == null) {
+                    return UnDefType.UNDEF;
+                }
+                return new DecimalType(goeResponse.forceState.toString());
+        }
+        return UnDefType.UNDEF;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            // we can not update single channels and refresh is triggered automatically
+            // anyways
+            return;
+        }
+
+        String key = null;
+        String value = null;
+
+        switch (channelUID.getId()) {
+            case MAX_CURRENT:
+                key = "amp";
+                if (command instanceof DecimalType) {
+                    value = String.valueOf(((DecimalType) command).intValue());
+                } else if (command instanceof QuantityType<?>) {
+                    value = String.valueOf(((QuantityType<ElectricCurrent>) command).toUnit(Units.AMPERE).intValue());
+                }
+                break;
+            case SESSION_CHARGE_CONSUMPTION_LIMIT:
+                key = "dwo";
+                var multiplier = 1000;
+                if (command instanceof DecimalType) {
+                    value = String.valueOf(((DecimalType) command).intValue() * multiplier);
+                } else if (command instanceof QuantityType<?>) {
+                    value = String.valueOf(
+                            ((QuantityType<Energy>) command).toUnit(Units.KILOWATT_HOUR).intValue() * multiplier);
+                }
+                break;
+            case PHASES:
+                key = "psm";
+                if (command instanceof DecimalType) {
+                    var phases = 1;
+                    var help = (DecimalType) command;
+                    if (help.intValue() == 3) {
+                        // set value 2 for 3 phases
+                        phases = 2;
+                    }
+                    value = String.valueOf(phases);
+                }
+                break;
+            case FORCE_STATE:
+                key = "frc";
+                if (command instanceof DecimalType) {
+                    value = String.valueOf(((DecimalType) command).intValue());
+                }
+        }
+
+        if (key != null && value != null) {
+            sendData(key, value);
+        } else {
+            logger.warn("Could not update channel {} with key {} and value {}", channelUID.getId(), key, value);
```

Comment:
See previous comment about thing state and logger level

Reply:
This is the situation if you try to set a key that is not allowed, the thing state will remain unchanged (ONLINE) then